github/github_test.go

```diff
@@ -105,6 +105,16 @@ func testHeader(t *testing.T, r *http.Request, header string, want string) {
 	}
 }
 
+func testCanonicalHeader(t *testing.T, r *http.Request, header string, want string) {
+	canonical := http.CanonicalHeaderKey(header)
+	for _, h := range r.Header[canonical] {
+		if h == want {
+			return
+		}
+	}
+	t.Errorf("Multiple header get, wanted %s but not found for header %s", want, header)
+}
```

Comment:
Can you explain why `testCanonicalHeader` is needed, instead of using `testHeader`?

It looks like its implementation differs from `testHeader` in that it checks all values for the `header` key rather than just the first like `testHeader` does, is that right? Is that neccessary? Does it make sense to add a new method rather than modify `testHeader` to check all header values?